### ISSUE TITLE
Bikin API baiknya berdasarkan UI Design atau CRUD table?

### ISSUE THREAD

**Issue by sahibul-nf** [ProgrammerZamanNow/qna#307]
Salam,

saya mengalami kasus ditempat kerja dimana API dibikin berdasarkan CRUD table, tidak atau bukan berdasarkan kebutuhan data untuk ditampilkan di UI aplikasi. Dalam artian, API response nya berupa bentuk dari table di database.

btw, apakah hal tersebut sesuatu yang bagus untuk dimplementasikan atau bagaimana ya klo jika dilihat dari segi peforma, produktifitas, kemudahan dll.

Terima kasih.

**Comment by febriliankr** [comment]
Kalau CRUD table berarti ada kemungkinan fetching data berlebihan di client, dan bisa jadi kalau mau ambil data dari 2 table harus 2 kalo network request ke API dong ya? Kedua hal ini bisa bikin request resources jauh lebih lambat.

Proses response yang sesuai kebutuhan di backend (pakai join table, atau pakai struktur data tertentu) jauh lebih cepet daripada harus multiple network request dari client.

**Comment by sahibul-nf** [comment]
Iya, jika butuh ambil data dari 2 table berarti harus 2 kali request ke API. 
Klo biasanya untuk get API detail product cukup sekali request, dengan kasus yg saya ceritakan jadi bisa lebih dari sekali request ke API.
Dan itu jadi cendrung jadi lebih kompleks implementasi nya di FE atau Mobile.
